### PR TITLE
v3(services): shared_spaces can include fields

### PR DESCRIPTION
**Pull request by kirederik**
Valid fields are:
* space: name, guid, relationships.organization
* space.organization: name, guid

[#171954418](https://www.pivotaltracker.com/story/show/171954418)

### CODE REVIEW

File: app/decorators/field_service_instance_organization_decorator.rb [hunk 1]
```diff
@@ -12,9 +12,9 @@ def initialize(fields)
       @fields = fields[:'space.organization'].to_set.intersection(self.class.allowed)
     end
 
-    def decorate(hash, service_instances)
+    def decorate(hash, resources)
       hash[:included] ||= {}
-      spaces = service_instances.map(&:space).uniq
+      spaces = resources.map { |r| r.try(:space) || r }.uniq
```
Comment (pivotal-marcela-campo) [inline app/decorators/field_service_instance_organization_decorator.rb:17]: I am not a fan of this in either of the two decorators. It looks to me more like accidental duplication and would have probably written them again as we might want to include different fields for space vs shared space/org vs shared org in the future. I am ok keeping this anyway, but would split them out as soon as they start deviating from each other.

File: docs/v3/source/includes/resources/service_instances/_list_shared_spaces.md.erb [hunk 2]
```diff
@@ -25,6 +25,19 @@ This endpoint lists the spaces that the service instance has been shared to.
 #### Definition
```
[the remark this answers is not in the document]
Comment (kirederik) [reply docs/v3/source/includes/resources/service_instances/_list_shared_spaces.md.erb:25]: Will add that, thanks!

File: app/decorators/field_service_instance_organization_decorator.rb (hunk 1 above)
Comment (kirederik) [reply app/decorators/field_service_instance_organization_decorator.rb:17]: Yeah, it's a bit weird, I agree. We can probably write something better that does not involve duplicating this code entirely.

I don't see the divergence of fields as a problem though. The message we create in the controller should determine what each method can include, returning earlier if the request includes non-allowed fields (that is, if the message is invalid).

You can see that the `relationships_shared_spaces`, `index` and `show` all use this same decorator, but not necessarily allow the same fields to be included.

Comment (pivotal-marcela-campo) [reply app/decorators/field_service_instance_organization_decorator.rb:17]: hmm, you are right. I always forget it is restricted in the message anyway. And also agree we can write something better than duplicating the whole code as well. The only thing I can think of at the moment is inheritance and having a get_spaces method overriden by subclasses. It feels overkilling though.

Comment (kirederik) [reply app/decorators/field_service_instance_organization_decorator.rb:17]: I'll merge this for now and we can always get back 👀

**Pull request merged**